sqlglot/dialects/e6.py

```diff
@@ -484,49 +529,138 @@ def _parse_cast(self, strict: bool, safe: t.Optional[bool] = None) -> exp.Expres
+                3.	Return Value:
+                    •	If no aggregates are found, the function returns an ArrayFilter object equivalent to:
+
+            exp.ArrayFilter(this=my_array, expression=lambda_expr)
+
+            """
+
+            def contains_aggregate(node: exp.Expression) -> bool:
+                """
+                Recursively checks if the given node or any of its children
+                contain an aggregate function.
+                """
+                if isinstance(node, exp.AggFunc):
                     return True
-                # traverse through all the children and check for the same
 
-                if not isinstance(root_node, exp.Expression):  # check if root_node is leaf node
+                if not isinstance(node, exp.Expression):
                     return False
 
-                child_nodes: dict = root_node.args
-                for key, value in child_nodes.items():
-                    contains_agg: bool = does_root_node_contain_AGG_expr(value)
-                    if contains_agg:
-                        return True
-                return False
+                # Recursively check all child nodes.
+                return any(contains_aggregate(child) for child in node.args.values())
+
+            # Retrieve the array expression and the lambda expression from the arguments.
+            array_expr = seq_get(self, 0)  # TODO:: Need to test this
+            lambda_expr = seq_get(self, 1)  # TODO:: Need to test this
 
-            if does_root_node_contain_AGG_expr(root_node):
-                # parser.Parser.raise_error(parser.Parser,message=
-                #                           f"Lambda expressions in filter functions are not supported in 'IN' clause or on aggregate functions")
+            # Get the root node of the lambda expression.
+            root_node = lambda_expr.args.get("this")
+
+            # If the lambda expression contains an aggregate function, raise a ValueError.
+            if contains_aggregate(root_node):
                 raise ValueError(
-                    "Lambda expressions in filter functions are not supported in 'IN' clause or on aggregate functions")
+                    "Lambda expressions in filter functions are not supported in 'IN' clause or on aggregate functions"
+                )
 
+            # Return an ArrayFilter expression with the parsed array and lambda expressions.
             return exp.ArrayFilter(this=array_expr, expression=lambda_expr)
 
-        def _parse_unnest_sql(self) -> exp.Expression:
-            array_expr = seq_get(self, 0)
-            # if (isinstance(array_expr, exp.Cast) and not exp.DataType.is_type(array_expr.to.this,
-            #                                                                   exp.DataType.Type.ARRAY)) or (
-            #         not isinstance(array_expr, exp.Array)):
-            #     raise ValueError(f"UNNEST function only supports array type")
+        @staticmethod
```

Comment:
File "/Users/adithyakanamarlapudi/PycharmProjects/e6-sqlglot/sqlglot/dialects/e6.py", line 777, in Parser
    "TO_TIMESTAMP": _build_datetime("TO_TIMESTAMP", exp.DataType.Type.TIMESTAMP),
TypeError: 'staticmethod' object is not callable


getting this error after chaning that to static method. It was working before

Reply:
@Adithyak-0926 let's connect for this